### PR TITLE
[Table] Show menu icon when menu is open in column header

### LINKED ISSUE #1297: Menu icon should always show when menu popover is open
### Bug report

- __Package version(s)__: `@blueprintjs/table` 1.18.0
- __Browser and OS versions__: macOS Sierra, latest Chrome

#### Steps to reproduce

1. Hover a column header
2. Click on menu icon to see menu popover
3. Hover away

#### Actual behavior

Menu is open but icon is gone:
image

#### Expected behavior

Icon always show when the menu is open.

### PR DESCRIPTION
**Pull request by llorca**
#### Screenshot

screen shot 2017-07-18 at 12 17 25 pm
screen shot 2017-07-18 at 12 13 43 pm

### CODE REVIEW

**Comment by cmslewis** [comment]
I noticed a style bug. Let's change the background color to `$header-selected-background-color` by default and `$header-hover-background-color` on hover.

File: packages/table/src/headers/columnHeaderCell.tsx [hunk 1]
```diff
@@ -98,6 +102,13 @@ export class ColumnHeaderCell extends AbstractComponent<IColumnHeaderCellProps,
             || target.classList.contains(Classes.TABLE_HEADER_CONTENT);
     }
 
+    public constructor() {
+        super();
+        this.state = {
+            isActive: false,
+        };
+    }
```
Comment (giladgray) [inline packages/table/src/headers/columnHeaderCell.tsx]: ❌ don't define constructor for static state, just do `public state = { isActive: false };` and be done with it. constructor is only needed if initial state depends on props.

Comment (giladgray) [inline packages/table/src/headers/columnHeaderCell.tsx]: also this constructor implementation is incorrect: must accept `(props, context?)` params. 
but really just remove it.

File: packages/table/src/headers/columnHeaderCell.tsx [hunk 2]
```diff
@@ -128,7 +139,7 @@ export class ColumnHeaderCell extends AbstractComponent<IColumnHeaderCellProps,
     }
 
     protected validateProps(nextProps: IColumnHeaderCellProps) {
-        if (nextProps.menu != null) {
+        if (nextProps != null && nextProps.menu != null) {
```
Comment (giladgray) [inline packages/table/src/headers/columnHeaderCell.tsx]: why add this check? nextProps should never be null; there are always props.

File: packages/table/src/headers/columnHeaderCell.tsx [hunk 3]
```diff
@@ -198,14 +209,17 @@ export class ColumnHeaderCell extends AbstractComponent<IColumnHeaderCellProps,
             pin: true,
             to: "window",
         }];
+        const classes = classNames(Classes.TABLE_TH_MENU_CONTAINER, {
+            [Classes.TABLE_TH_MENU_OPEN]: this.state.isActive,
```
Comment (giladgray) [inline packages/table/src/headers/columnHeaderCell.tsx]: when does `this.state.isActive` change? i don't see any `setState`s in this PR...

File: packages/table/src/headers/columnHeaderCell.tsx (hunk 2 above)
Comment (llorca) [inline packages/table/src/headers/columnHeaderCell.tsx]: It was undefined (with console error) and this check fixed it. I'll investigate more tomorrow and get back to you

File: packages/table/src/headers/columnHeaderCell.tsx (hunk 3 above)
Comment (llorca) [inline packages/table/src/headers/columnHeaderCell.tsx]: The component was _setting_ its state already, even though there was not state _defined_--check the very bottom of the file outside of this PR. Oversight from previous work on this file?

File: packages/table/src/headers/columnHeaderCell.tsx (hunk 1 above)
Comment (llorca) [inline packages/table/src/headers/columnHeaderCell.tsx]: Will do.

File: packages/table/src/headers/columnHeaderCell.tsx (hunk 3 above)
Comment (giladgray) [inline packages/table/src/headers/columnHeaderCell.tsx]: oh wow ok 🆒

File: packages/table/src/headers/columnHeaderCell.tsx (hunk 2 above)
Comment (llorca) [inline packages/table/src/headers/columnHeaderCell.tsx]: Works now, might have been because of the constructor? I dunno.

**Comment by llorca** [comment]
Fixed the visual bugs (nice catch @cmslewis, so much better now) and addressed comments 👍

**Comment by llorca** [comment]
@giladgray ready for review!

**Pull request merged**